src/App.tsx

```diff
@@ -17,18 +42,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={initialTodos}
+                onSetVisibleTodos={setVisibleTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              <TodoList
+                todos={visibleTodos}
+                onSelectTodo={selectTodo}
+                onSetModalVisibility={setModalVisbility}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalVisible && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          currentUser={user}
+          onResetTodo={selectTodo}
+          onResetUser={setUser}
+          onResetModalVisibility={setModalVisbility}
```

Comment:
if you only use it together, you can combine it in a single function and pass it as a prop